src/flint/fmpz.jl

```diff
@@ -1267,6 +1267,37 @@ function divisible(x::fmpz, y::Int)
               (Ref{fmpz}, Int), x, y))
 end
 
+@doc Markdown.doc"""
+    divisors(a::fmpz)
+
+Return the positive divisors of $a$ in an array, not necessarily in growing
+order. We require $a \neq 0$.
+"""
+function divisors(a::fmpz)
+   iszero(a) && throw(DomainError("Argument must be non-zero"))
+
+   divs = fmpz[one(FlintZZ)]
+   isone(a) && return divs
+
+   for (p,e) in factor(a)
+      ndivs = copy(divs)
+      for i = 1:e
+         map!(d -> p*d, ndivs, ndivs)
+         append!(divs, ndivs)
+      end
+   end
+
+   return divs
+end
+
+@doc Markdown.doc"""
+    divisors(a::Int)
+
+Return the positive divisors of $a$ in an array, not necessarily in growing
+order. We require $a \neq 0$.
+"""
```

Comment:
I would suggest not duplicating the docstring, e.g. by attaching it to the function:
```
@doc Markdown.doc"""
    divisors(a::Union{Int,fmpz})
Return the positive divisors of $a$ in an array, not necessarily in growing
order. We require $a \neq 0$.
"""
function divisors end
```

Reply:
Isn't this similair to what was written here https://github.com/Nemocas/AbstractAlgebra.jl/pull/732#pullrequestreview-570422311 ? I'm all for doing this, given that this is consistent with the rest of the code.

Reply:
There is an issue #535 about not duplicating docstrings which has some support, so it's better to not add more duplicates (which represent more work to deduplicate them later).

Reply:
I did not think of this creating such big issues later on -- this will help me on my current work. Thank you very much!